### ISSUE TITLE
Rancher 2.4.2 in k3s on Fedora31 will not start -> error creating binary drop folder: mkdir ./management-state: permission denied

### ISSUE THREAD

**Issue by lukasmrtvy** [rancher/rancher#26596]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
- Fedora31
- k3s
- Rancher 2.4.2 Helm chart

**Result:**
Rancher will not start due:
```
2020/04/10 13:55:11 [INFO] Listening on :80
2020/04/10 13:55:11 [ERROR] UserAttributeController user-2xll8 [mgmt-auth-userattributes-controller] failed with : refresh daemon not yet initialized
2020/04/10 13:55:11 [INFO] Refreshing driverMetadata in 1440 minutes
2020/04/10 13:55:11 [INFO] Active TLS secret serving-cert (ver=580696) (count 12): map[field.cattle.io/projectId:local:p-6b9t7 listener.cattle.io/cn-10.42.0.40:10.42.0.40 listener.cattle.io/cn-10.42.0.41:10.42.0.41 listener.cattle.io/cn-10.42.0.42:10.42.0.42 listener.cattle.io/cn-10.42.0.43:10.42.0.43 listener.cattle.io/cn-10.42.0.44:10.42.0.44 listener.cattle.io/cn-10.42.0.45:10.42.0.45 listener.cattle.io/cn-10.42.0.46:10.42.0.46 listener.cattle.io/cn-10.42.0.57:10.42.0.57 listener.cattle.io/cn-10.42.0.63:10.42.0.63 listener.cattle.io/cn-127.0.0.1:127.0.0.1 listener.cattle.io/cn-localhost:localhost listener.cattle.io/hash:dc1a579a1a98f55f77ec474b4e7f22bc440115ea101ad30d6fa4b5cc54266c27]
2020/04/10 13:55:11 [ERROR] CatalogController helm3-library [catalog] failed with : mkdir management-state: permission denied
2020/04/10 13:55:11 [ERROR] CatalogController library [catalog] failed with : mkdir management-state: permission denied
2020/04/10 13:55:11 [ERROR] CatalogController system-library [catalog] failed with : mkdir management-state: permission denied
2020/04/10 13:55:11 [INFO] Starting /v1, Kind=Secret controller
2020/04/10 13:55:11 [INFO] Reconciling GlobalRoles
2020/04/10 13:55:11 [INFO] Starting management.cattle.io/v3, Kind=Cluster controller
2020/04/10 13:55:11 [INFO] Active TLS secret serving-cert (ver=580696) (count 12): map[field.cattle.io/projectId:local:p-6b9t7 listener.cattle.io/cn-10.42.0.40:10.42.0.40 listener.cattle.io/cn-10.42.0.41:10.42.0.41 listener.cattle.io/cn-10.42.0.42:10.42.0.42 listener.cattle.io/cn-10.42.0.43:10.42.0.43 listener.cattle.io/cn-10.42.0.44:10.42.0.44 listener.cattle.io/cn-10.42.0.45:10.42.0.45 listener.cattle.io/cn-10.42.0.46:10.42.0.46 listener.cattle.io/cn-10.42.0.57:10.42.0.57 listener.cattle.io/cn-10.42.0.63:10.42.0.63 listener.cattle.io/cn-127.0.0.1:127.0.0.1 listener.cattle.io/cn-localhost:localhost listener.cattle.io/hash:dc1a579a1a98f55f77ec474b4e7f22bc440115ea101ad30d6fa4b5cc54266c27]
2020/04/10 13:55:11 [INFO] Reconciling RoleTemplates
2020/04/10 13:55:11 [INFO] driverMetadata: refreshing data from upstream https://releases.rancher.com/kontainer-driver-metadata/release-v2.4/data.json
panic: error creating binary drop folder: mkdir ./management-state: permission denied

goroutine 1874 [running]:
github.com/rancher/rancher/app.(*Rancher).Start.func1(0x46e8a60, 0xc001e87e00)
        /go/src/github.com/rancher/rancher/app/app.go:245 +0x651
created by github.com/rancher/rancher/vendor/github.com/rancher/wrangler/pkg/leader.run.func1
        /go/src/github.com/rancher/rancher/vendor/github.com/rancher/wrangler/pkg/leader/leader.go:58 +0x46
```
**Other details that may be helpful:**

**Environment information**
- Rancher version: `rancher/rancher:v2.4.2`
- Installation option: `single node`

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
`custom`
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
`metal (proxmox)`
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4+k3s1", GitCommit:"3eee8ac3a1cf0a216c8a660571329d4bda3bdf77", GitTreeState:"clean", BuildDate:"2020-03-25T16:13:25Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4+k3s1", GitCommit:"3eee8ac3a1cf0a216c8a660571329d4bda3bdf77", GitTreeState:"clean", BuildDate:"2020-03-25T16:13:25Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```
- Docker version (use `docker version`):
```
none
```

EDIT:
I have installed https://github.com/rancher/k3s-selinux, but seems these policies are not enough, getting:
```
 denied  { create } for  pid=319353 comm="rancher" name="management-state" scontext=system_u:system_r:container_t:s0:c164,c1003 tcontext=system_u:object_r:container_var_lib_t:s0 tclass=dir permissive=0
```

**Comment by cjellick** [comment]
Need a little more info (for starters) you said `Rancher 2.4.2 Helm chart` but then also said `Installation option: single node`.
These two are contradictory. Single node option means you launched rancher via `docker run -d ... rancher/rancher:v2.4.2`.
Helm chart means you created a Kubernetes cluster and then used the helm chart to launch Rancher within that cluster.

Which did you do?

**Comment by lukasmrtvy** [comment]
k3s cluster and helm chart for rancher

**Comment by cjellick** [comment]
Did you also install container-selinux selinux-policy-base according to
https://rancher.com/docs/k3s/latest/en/advanced/#experimental-selinux-support ?

**Comment by lukasmrtvy** [comment]
yes

**Comment by sangnguyen7** [comment]
I got the same issue. I'm install using Helm Chart on v2.4.2 on K3S in Rhel 7 enviroment.

```
2020/04/17 04:04:09 [INFO] Refreshing all schemas
2020/04/17 04:04:09 [INFO] Starting management controllers
2020/04/17 04:04:10 [INFO] Refreshing driverMetadata in 1440 minutes
2020/04/17 04:04:10 [ERROR] ClusterController local [cluster-deploy] failed with : waiting for server-url setting to be set
2020/04/17 04:04:10 [ERROR] CatalogController system-library [catalog] failed with : mkdir management-state: permission denied
2020/04/17 04:04:10 [ERROR] CatalogController helm3-library [catalog] failed with : mkdir management-state: permission denied
2020/04/17 04:04:10 [INFO] Reconciling GlobalRoles
2020/04/17 04:04:10 [INFO] Starting management.cattle.io/v3, Kind=Cluster controller
2020/04/17 04:04:10 [ERROR] CatalogController library [catalog] failed with : mkdir management-state: permission denied
2020/04/17 04:04:10 [INFO] Reconciling RoleTemplates
2020/04/17 04:04:10 [INFO] driverMetadata: refreshing data from upstream https://releases.rancher.com/kontainer-driver-metadata/release-v2.4/data.json
2020/04/17 04:04:10 [ERROR] CatalogController system-library [catalog] failed with : mkdir management-state: permission denied
2020/04/17 04:04:10 [ERROR] ClusterController local [cluster-deploy] failed with : waiting for server-url setting to be set
2020/04/17 04:04:10 [ERROR] CatalogController library [catalog] failed with : mkdir management-state: permission denied
2020/04/17 04:04:10 [ERROR] CatalogController helm3-library [catalog] failed with : mkdir management-state: permission denied
panic: error creating binary drop folder: mkdir ./management-state: permission denied

goroutine 2168 [running]:
github.com/rancher/rancher/app.(*Rancher).Start.func1(0x46e8a60, 0xc00111f200)
        /go/src/github.com/rancher/rancher/app/app.go:245 +0x651
created by github.com/rancher/rancher/vendor/github.com/rancher/wrangler/pkg/leader.run.func1
        /go/src/github.com/rancher/rancher/vendor/github.com/rancher/wrangler/pkg/leader/leader.go:58 +0x46
```

**Comment by lukasmrtvy** [comment]
I created issue https://github.com/rancher/k3s-selinux/issues/7

**Comment by mfld** [comment]
Hello,

We are seeing this issue on centos 8
NAME="CentOS Linux"
VERSION="8"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"
CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"

Trying to create or remove a cluster in rancher web-UI hans forever with errors:
Failed while: Wait for Condition: InitialRolesPopulated: True

Reviewing the rancher logs:
rancher-688b6d8c67-8xhd6 rancher 2021/02/01 07:18:12 [ERROR] failed to call leader func: failed to add management data: error creating binary drop folder: mkdir ./management-state: permission denied

/best regards

**Comment by jonaz** [comment]
Adding more info to the centos 8 issue above. We are getting this auditbeat log from selinux when the error happens. 
```
{
  "_index": "long-auditbeat-7.10.0-2021.02",
  "_type": "_doc",
  "_id": "zB9-XHcBJ5isfxxD0E4H",
  "_version": 1,
  "_score": null,
  "_source": {
    "service": {
      "type": "auditd"
    },
    "ecs": {
      "version": "1.5.0"
    },
    "auditd": {
      "data": {
        "seperms": "write",
        "seresult": "denied",
        "dev": "dm-6",
        "tclass": "dir",
        "scontext": "system_u:system_r:container_t:s0:c390,c774",
        "a0": "ffffffffffffff9c",
        "syscall": "mkdirat",
        "ino": "34858125",
        "tty": "(none)",
        "arch": "x86_64",
        "exit": "EACCES",
        "a2": "1ff",
        "a3": "c0057fc880",
        "name": "75b544fd5e3501a9d4aecb5357cf4ea9215198804e01cf94dc398e45798f4384",
        "permissive": "0",
        "tcontext": "system_u:object_r:container_ro_file_t:s0",
        "a1": "c0057fc880"
      },
      "result": "fail",
      "paths": [
        {
          "rdev": "00:00",
          "cap_fe": "0",
          "dev": "fd:06",
          "cap_fver": "0",
          "cap_fi": "0",
          "mode": "040755",
          "ouid": "1000",
          "item": "0",
          "ogid": "0",
          "obj_user": "system_u",
          "cap_frootid": "0",
          "name": "./",
          "inode": "34858125",
          "nametype": "PARENT",
          "cap_fp": "0",
          "obj_role": "object_r",
          "obj_domain": "container_ro_file_t",
          "obj_level": "s0"
        },
        {
          "cap_fe": "0",
          "cap_frootid": "0",
          "cap_fver": "0",
          "cap_fi": "0",
          "item": "1",
          "name": "./management-state",
          "cap_fp": "0",
          "nametype": "CREATE"
        }
      ],
      "message_type": "avc",
      "sequence": 726019,
      "summary": {
        "actor": {
          "primary": "unset",
          "secondary": "root"
        },
        "how": "/usr/bin/rancher",
        "object": {
          "type": "policy"
        }
      }
    },
    "tags": [
      "beats_input_raw_event",
      "long"
    ],
    "agent": {
      "hostname": "k8s-rancher002.sto2.fnox.se",
      "name": "k8s-rancher002.sto2.fnox.se",
      "version": "7.10.0",
      "id": "1d10c795-772f-4a5a-b371-df52033e67db",
      "ephemeral_id": "bb035d56-787f-4c43-961b-7fc106f6fde9",
      "type": "auditbeat"
    },
    "event": {
      "module": "auditd",
      "outcome": "failure",
      "category": [
        "file"
      ],
      "action": "violated-apparmor-policy",
      "kind": "event",
      "type": [
        "creation"
      ]
    },
    "user": {
      "group": {
        "id": "0",
        "name": "root"
      },
      "name": "root",
      "selinux": {
        "category": "c390,c774",
        "domain": "container_t",
        "level": "s0",
        "role": "system_r",
        "user": "system_u"
      },
      "filesystem": {
        "id": "0",
        "group": {
          "id": "0",
          "name": "root"
        },
        "name": "root"
      },
      "id": "0",
      "effective": {
        "id": "0",
        "name": "root",
        "group": {
          "id": "0",
          "name": "root"
        }
      },
      "saved": {
        "id": "0",
        "name": "root",
        "group": {
          "id": "0",
          "name": "root"
        }
      }
    },
    "@version": "1",
    "host": {
      "name": "k8s-rancher002.sto2.fnox.se"
    },
    "@timestamp": "2021-02-01T07:29:09.465Z",
    "type": "audit",
    "process": {
      "title": "rancher --http-listen-port=80 --https-listen-port=443 --audit-log-path=/var/log/auditlog/rancher-api-audit.log --audit-level=0 -",
      "name": "rancher",
      "ppid": 2124316,
      "executable": "/usr/bin/rancher",
      "working_directory": "/var/lib/rancher",
      "pid": 2124358
    }
  },
  "fields": {
    "@timestamp": [
      "2021-02-01T07:29:09.465Z"
    ]
  },
  "highlight": {
    "agent.hostname": [
      "@kibana-highlighted-field@k8s@/kibana-highlighted-field@-@kibana-highlighted-field@rancher002@/kibana-highlighted-field@.@kibana-highlighted-field@sto2@/kibana-highlighted-field@.@kibana-highlighted-field@fnox.se@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1612164549465
  ]
}
```

**Comment by jonaz** [comment]
Did some more debugging inside rancher container:
```
root@rancher-6686db677b-hlppb:/var/lib/rancher# ls -laZ
total 0
drwxr-xr-x. 5 rancher root system_u:object_r:container_ro_file_t:s0        53 Feb  3 13:15 .
drwxr-xr-x. 1 root    root system_u:object_r:container_file_t:s0:c325,c670 52 Feb  3 13:15 ..
drwxr-xr-x. 2 rancher root system_u:object_r:container_ro_file_t:s0         6 Jan  9 01:59 etcd
drwxr-xr-x. 3 root    root system_u:object_r:container_ro_file_t:s0        19 Feb  3 13:15 k3s
drwx------. 2 root    root system_u:object_r:container_ro_file_t:s0        18 Feb  3 13:15 management-state
```

The problem here is that the context is `container_ro_file_t` here so `./management-state` can never be created when running selinux enforcing?

**Comment by jonaz** [comment]
` reinstall rke2-selinux` fixes the issue in runtime. Rebooting the server will restart the pod with wrong context again `container_ro_file_t`?

**Comment by lukasmrtvy** [comment]
@jonaz use edit please,, instead of adding new comments...

**Comment by mfld** [comment]
https://github.com/rancher/rke2/issues/687#issuecomment-774306048 related?

**Comment by davidnuzik** [comment]
Related: https://github.com/rancher/rke2/issues/687

**Comment by davidnuzik** [comment]
Related: https://github.com/rancher/rke2/issues/690

**Comment by Wolskie** [comment]
I am also experiencing this issue:
I have a strong feeling it is related to this https://github.com/containerd/containerd/issues/5090

**Comment by jonaz** [comment]
It was resolved in v1.19.9+rke2r1 when we upgraded the error went away.

**Comment by Wolskie** [comment]
> It was resolved in v1.19.9+rke2r1 when we upgraded the error went away.

Very interesting, we're currently running `v1.20.4+rke2r1` with this problem.
